fix(NcAppSidebar): make sidebar a single node to allow using v-show

ShGKme opened the pull request:
### ☑️ Resolves

- Fix https://github.com/nextcloud-libraries/nextcloud-vue/issues/5623

### 🖼️ Screenshots

🏚️ Before | 🏡 After
---|---
B | A

### 🚧 Tasks

- [ ] ...

### 🏁 Checklist

- [ ] ⛑️ Tests are included or are not applicable
- [ ] 📘 Component documentation has been extended, updated or is not applicable
- [ ] 3️⃣ Backport to `next` requested with a Vue 3 upgrade

susnux commented:
I vote for not backporting this to next and instead require to use `open` prop. Maybe introduce `no-toggle` prop to hide the toggle button.

susnux reviewed (commented):
Do you prefer this one? Because I think it adds a lot of complexity.
If yes I will free some time for a review :sweat_smile:

ShGKme commented:
> Do you prefer this one?

I dislike both.

I think this is much simpler (check diff with hide whitespaces and without looking into docs — it's literally just removing `<Fragment>` and adding `<Portal>`).

But this could be more risky because we need to rely on `#content-vue` or another selector and because Teleport to a node inside Virtual DOM is always risky.

susnux commented on src/components/NcAppSidebar/NcAppSidebar.vue:
```diff
@@ -451,177 +448,265 @@ export default {
+				So we cannot render the toggle button directly in this component.
+				As a simple solution - render it in the content to keep correct position.
+			-->
+			<Teleport selector="#content-vue">
```
We should add `no-toggle` prop to allow hiding the toggle button.

susnux replied:
but unrelated, just to note.

ShGKme replied:
I can restore a PR with a feature of placing `NcAppSidebarToggle` button anywhere and `no-toggle` prop.

ShGKme replied:
Only `no-toggle`: https://github.com/nextcloud-libraries/nextcloud-vue/pull/5702

susnux commented on src/components/NcAppSidebar/NcAppSidebar.vue:
```diff
@@ -1151,7 +1235,6 @@ $top-buttons-spacing: $app-navigation-padding; // align with app navigation
 */
 .app-sidebar {
 	--app-sidebar-width: clamp(300px, 27vw, 500px);
-	--app-sidebar-padding: #{$app-navigation-padding};
```
Dont we need to style the toggle?

susnux replied:
Why removing this?

susnux reviewed (approved):
works.

I am fine with both solutions :shrug:

ShGKme commented:
- Rebased onto `master` to resolve conflicts
- Added a small `fixup`:
  - Removed `inheritAttrs: false` (not needed anymore)
  - Moved `v-if` from `<NcButton>` to `<teleport>` to not have unneeded teleport without button.

Antreesy reviewed (approved):
I'd personnaly prefer this solution (not a fan of several nodes under root template, now it's more predictable).
Tested, works fine as a middleground before moving to `open.sync`

susnux commented on src/components/NcAppSidebar/NcAppSidebar.vue:
```diff
@@ -1060,6 +1156,19 @@ export default {
+		checkToggleButtonContainerAvailability() {
+			// Toggle button must be rendered, but there is no element to teleport it to
+			if (this.open === false && !this.noToggle && !this.ncContentSelector) {
+				console.warn(
```
why not Vue.warn ?

ShGKme replied:
I'd keep it to have warning also in the production build of an app.

But we should definitely depside what we should use in which case for warnings in frontend.

ShGKme commented:
Rebased onto master and squashed

susnux commented:
@ShGKme backport?
